src/controllTools.js

```diff
@@ -0,0 +1,67 @@
+import { checkbox, editTask } from './index.js';
+import { getTask } from './data.js';
+
+export const deleteTask = () => {
+  const clearTask = document.querySelector('.clear-task');
+  clearTask.addEventListener('click', () => {
+    if (getTask().length > 0) {
+      const filterCompliteTask = getTask().filter(
+        (task) => task.completed !== true,
+      );
+      localStorage.setItem('Task-list', JSON.stringify(filterCompliteTask));
+      window.location.reload();
+    }
+  });
+};
+
+export const addTask = () => {
+  class Task {
+    constructor(description) {
+      this.description = description;
+      this.index = new Date();
+      this.completed = false;
+    }
+  }
+  const addTaskForm = document.querySelector('#add-task');
+  const data = JSON.parse(localStorage.getItem('Task-list')) || [];
+  addTaskForm.addEventListener('click', () => {
+    const inputTaskValue = document.querySelector('#add-task-input').value;
+    if (inputTaskValue === '') return;
+    const newTask = new Task(inputTaskValue);
+    data.push(newTask);
+    localStorage.setItem('Task-list', JSON.stringify(data));
+  });
+  return data;
+};
+
+export const deleteOne = (deleteIcon, taskId) => {
+  deleteIcon.addEventListener('click', () => {
+    const filteredTask = getTask().filter((task) => task.index !== taskId);
+    localStorage.setItem('Task-list', JSON.stringify(filteredTask));
+    window.location.reload();
+  });
+};
+
+export const updateTask = () => {
+  checkbox.forEach((check) => {
+    check.addEventListener('click', () => {
+      editTask.forEach((task) => {
+        if (check.id === task.dataset.indexNumber) {
+          getTask().forEach((item) => {
+            if (item.index === check.id) {
+              if (item.completed) {
+                item.completed = false;
+                localStorage.setItem('Task-list', JSON.stringify(getTask()));
+                window.location.reload();
+              } else {
+                item.completed = true;
+                localStorage.setItem('Task-list', JSON.stringify(getTask()));
+                window.location.reload();
+              }
```

Comment:
I love the structure of instructions followed here 👏 , all of the checks for the task id and checking if it's completed is perfect 👍 . I just had an idea that it could be made simpler following a KISS methodology instead of having nested if checks and having an else statement that could be completely avoided. you could do something like the suggested change in green below:
```suggestion
            if (item.index === check.id) {
                item.completed = item.completed ? false : true;
                localStorage.setItem('Task-list', JSON.stringify(getTask()));
                window.location.reload();
              }           
```
This makes the check simpler with lesser code than the original and it still works the same way. for example, if item. completed is true it is checked and set as false, if it's false it is checked and set as true.

Reply:
This method is simple but instead `item.completed = item.completed ? false : true;` I put `item.completed = !item.completed;`. Thanks for your feedback :pray:

Reply:
Yes, this idea is even smarter. good Job 👏 .